test/jasmine/tests/plots_test.js

```diff
@@ -427,4 +427,71 @@ describe('Test Plots', function() {
             expect(gd._transitioning).toBeUndefined();
         });
     });
+
+    describe('extendObjectWithContainers', function() {
+
+        function assert(dest, src, expected) {
+            Plots.extendObjectWithContainers(dest, src, ['container']);
+            expect(dest).toEqual(expected);
+        }
+
+        it('extend each container items', function() {
+            var dest = {
+                container: [
+                    { text: '1', x: 1, y: 1 },
+                    { text: '2', x: 2, y: 2 }
+                ]
+            };
+
+            var src = {
+                container: [
+                    { text: '1-new' },
+                    { text: '2-new' }
+                ]
+            };
+
+            var expected = {
+                container: [
+                    { text: '1-new', x: 1, y: 1 },
+                    { text: '2-new', x: 2, y: 2 }
+                ]
+            };
+
+            assert(dest, src, expected);
+        });
+
+        it('clears container items when applying null src items', function() {
+            var dest = {
+                container: [
+                    { text: '1', x: 1, y: 1 },
+                    { text: '2', x: 2, y: 2 }
+                ]
+            };
+
+            var src = {
+                container: [null, null]
+            };
+
+            var expected = {
+                container: [null, null]
```

Comment:
N.B. I changed the behavior here as discussed in https://github.com/plotly/plotly.js/pull/1118#discussion_r87019878 - see https://github.com/plotly/plotly.js/pull/1118/commits/36859f10900709d1beee28cb8c314f00d1c5b015

In brief, frames with array containers set to null extend the state with `null`. It is now up to the subsequent `supplyDefaults` calls to coerced those `null` items into `{}`, as in regular `Plotly.plot` calls.